src/pubtools/sign/signers/msgsigner.py

```diff
@@ -0,0 +1,434 @@
+from __future__ import annotations
+
+import base64
+from dataclasses import field, dataclass, asdict
+import json
+import logging
+from typing import Dict, List, ClassVar, Any, Optional
+import uuid
+import os
+
+import OpenSSL
+import click
+
+from . import Signer
+from ..operations.base import SignOperation
+from ..operations import ClearSignOperation, ContainerSignOperation
+from ..results.signing_results import SigningResults
+from ..results import ClearSignResult, ContainerSignResult
+from ..results import SignerResults
+from ..exceptions import UnsupportedOperation
+from ..clients.msg_send_client import SendClient
+from ..clients.msg_recv_client import RecvClient
+from ..models.msg import MsgMessage
+from ..conf.conf import load_config, CONFIG_PATHS
+from ..utils import set_log_level, isodate_now
+
+
+LOG = logging.getLogger("pubtools.sign.signers.msgsigner")
+
+
+@dataclass()
+class MsgSignerResults(SignerResults):
+    """MsgSignerResults model."""
+
+    status: str
+    error_message: str
+
+    def to_dict(self: SignerResults):
+        """Return dict representation of MsgSignerResults model."""
+        return {"status": self.status, "error_message": self.error_message}
+
+
+@dataclass()
+class MsgSigner(Signer):
```

Comment:
Why is this a data class? It seems to be doing actual operations.

Reply:
Should every Python class just become dataclass then? My understanding of data classes is that they primarily exist to hold data and perform transformation operations on that data. If networking is involved then it's not appropriate to use them. PEP generally agrees: https://peps.python.org/pep-0557/#rationale .

Reply:
Where exactly in the PEP you see that? I see:
- The decorated classes are truly “normal” Python classes. The Data Class decorator should not interfere with any usage of the class.
-  One main design goal of Data Classes is to support static type checkers. 

There isn't any mention dataclasses should be use only to support model classes or any mention that they shouldn't be used to anything else than model classes. I used dataclass for the signer as it reads the configuration some input. It also enhance type checking. And for the last it allows you to skip init definition

Reply:
"There have been numerous attempts to define classes which exist primarily to store values which are accessible by attribute lookup"